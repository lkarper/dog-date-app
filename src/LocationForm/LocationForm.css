.LocationForm__fieldset {
    width: 100%;
}