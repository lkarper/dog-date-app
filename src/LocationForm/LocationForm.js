import React, { useState, useEffect } from 'react';
import ValidateAddress from '../validation-components/create-howl-validation/ValidateAddress';
import ValidateCity from '../validation-components/create-howl-validation/ValidateCity';
import StateSelector from '../StateSelector/StateSelector';
import ValidateState from '../validation-components/create-howl-validation/ValidateState';
import ValidateZipCode from '../validation-components/create-howl-validation/ValidateZipCode';
import './LocationForm.css';

const LocationForm = (props) => {

    const { setLocation, setLocationError, location } = props;

    const [address, setAddress] = useState(location.address);
    const [addressError, setAddressError] = useState('');
    const [city, setCity] = useState(location.city);
    const [cityError, setCityError] = useState('');
    const [state, setState] = useState(location.state);
    const [stateError, setStateError] = useState('');
    const [zipcode, setZipcode] = useState(location.zipcode);
    const [zipcodeError, setZipcodeError] = useState('');

    useEffect(() => {
        const errorsArray = [];

        if (addressError) {
            errorsArray.push(addressError);
        }
        
        if (cityError) {
            errorsArray.push(cityError);
        }

        if (stateError) {
            errorsArray.push(stateError);
        }

        if (zipcodeError) {
            errorsArray.push(zipcodeError);
        }

        setLocationError(errorsArray);

    }, [addressError, cityError, stateError, zipcodeError, setLocationError]);

    useEffect(() => {
        setLocation({
            address,
            city,
            state,
            zipcode
        });
    }, [address, city, state, zipcode, setLocation]);

    return (
        <fieldset className='LocationForm__fieldset'>
            <legend>Enter the location for your meeting place here</legend>
            <div>
                <label htmlFor='address'>Location or street address:</label>
                <input 
                    type="text" 
                    id="address" 
                    name="address"
                    value={address}
                    aria-describedby='address-validator'
                    onChange={(e) => setAddress(e.target.value)} 
                    required
                />
            </div>
            <div role='alert'>
                <ValidateAddress 
                    address={address}
                    addressError={addressError}
                    setAddressError={setAddressError}
                />
            </div>
            <div>
                <label htmlFor='city'>City:</label>
                <input 
                    type='text'
                    id='city'
                    name='city'
                    value={city}
                    aria-describedby='city-validator'
                    onChange={(e) => setCity(e.target.value)}
                    required
                />
            </div>
            <div role='alert'>
                <ValidateCity 
                    city={city}
                    cityError={cityError}
                    setCityError={setCityError}
                />
            </div>
            <StateSelector 
                propState={state}
                setState={setState}
            />
            <div role='alert'>
                <ValidateState 
                    state={state}
                    stateError={stateError}
                    setStateError={setStateError}
                />
            </div>
            <div>
                <label htmlFor='zipcode'>ZIP code:</label>
                <input 
                    type='text'
                    id='zipcode'
                    name='zipcode'
                    value={zipcode}
                    aria-describedby='zipcode-validator'
                    onChange={(e) => setZipcode(e.target.value)}
                    required
                />
            </div>
            <div role='alert'>
                <ValidateZipCode 
                    zipcode={zipcode}
                    zipcodeError={zipcodeError}
                    setZipcodeError={setZipcodeError}
                />
            </div>
        </fieldset>
    );
}

export default LocationForm;
