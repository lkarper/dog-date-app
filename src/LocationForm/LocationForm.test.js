import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LocationForm from './LocationForm';

describe('LocationForm component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LocationForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<LocationForm />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
