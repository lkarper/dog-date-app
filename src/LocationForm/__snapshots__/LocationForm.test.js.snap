// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationForm component renders the UI as expected 1`] = `
<LocationForm
  location={
    Object {
      "address": "",
      "city": "",
      "state": "",
      "zipcode": "",
    }
  }
  setLocation={[Function]}
  setLocationError={[Function]}
>
  <fieldset
    className="LocationForm__fieldset sub-fieldset"
  >
    <legend>
      Enter the location for your meeting place here
    </legend>
    <div>
      <label
        htmlFor="address"
      >
        Location or street address:
         
      </label>
      <input
        aria-describedby="address-validator"
        id="address"
        maxLength="100"
        name="address"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="LocationForm__alert-div large"
      role="alert"
    >
      <ValidateAddress
        address=""
        addressError="You must enter an address or location name. (Max 100 chars.)"
        setAddressError={[Function]}
      >
        <p
          className="ValidateAddress__validator error"
          id="address-validator"
        >
          You must enter an address or location name. (Max 100 chars.)
        </p>
      </ValidateAddress>
    </div>
    <div>
      <label
        htmlFor="city"
      >
        City:
         
      </label>
      <input
        aria-describedby="city-validator"
        id="city"
        maxLength="100"
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="LocationForm__alert-div large"
      role="alert"
    >
      <ValidateCity
        city=""
        cityError="You must enter a city. (Max 100 chars.)"
        setCityError={[Function]}
      >
        <p
          className="ValidateCity__validator error"
          id="city-validator"
        >
          You must enter a city. (Max 100 chars.)
        </p>
      </ValidateCity>
    </div>
    <StateSelector
      propState=""
      setState={[Function]}
    >
      <div>
        <label
          htmlFor="state"
        >
          State:
        </label>
        <select
          aria-describedby="state-validator"
          id="state"
          name="state"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value=""
          >
            Select a state:
          </option>
          <option
            value="AL"
          >
            Alabama
          </option>
          <option
            value="AK"
          >
            Alaska
          </option>
          <option
            value="AZ"
          >
            Arizona
          </option>
          <option
            value="AR"
          >
            Arkansas
          </option>
          <option
            value="CA"
          >
            California
          </option>
          <option
            value="CO"
          >
            Colorado
          </option>
          <option
            value="CT"
          >
            Connecticut
          </option>
          <option
            value="DE"
          >
            Delaware
          </option>
          <option
            value="DC"
          >
            District Of Columbia
          </option>
          <option
            value="FL"
          >
            Florida
          </option>
          <option
            value="GA"
          >
            Georgia
          </option>
          <option
            value="HI"
          >
            Hawaii
          </option>
          <option
            value="ID"
          >
            Idaho
          </option>
          <option
            value="IL"
          >
            Illinois
          </option>
          <option
            value="IN"
          >
            Indiana
          </option>
          <option
            value="IA"
          >
            Iowa
          </option>
          <option
            value="KS"
          >
            Kansas
          </option>
          <option
            value="KY"
          >
            Kentucky
          </option>
          <option
            value="LA"
          >
            Louisiana
          </option>
          <option
            value="ME"
          >
            Maine
          </option>
          <option
            value="MD"
          >
            Maryland
          </option>
          <option
            value="MA"
          >
            Massachusetts
          </option>
          <option
            value="MI"
          >
            Michigan
          </option>
          <option
            value="MN"
          >
            Minnesota
          </option>
          <option
            value="MS"
          >
            Mississippi
          </option>
          <option
            value="MO"
          >
            Missouri
          </option>
          <option
            value="MT"
          >
            Montana
          </option>
          <option
            value="NE"
          >
            Nebraska
          </option>
          <option
            value="NV"
          >
            Nevada
          </option>
          <option
            value="NH"
          >
            New Hampshire
          </option>
          <option
            value="NJ"
          >
            New Jersey
          </option>
          <option
            value="NM"
          >
            New Mexico
          </option>
          <option
            value="NY"
          >
            New York
          </option>
          <option
            value="NC"
          >
            North Carolina
          </option>
          <option
            value="ND"
          >
            North Dakota
          </option>
          <option
            value="OH"
          >
            Ohio
          </option>
          <option
            value="OK"
          >
            Oklahoma
          </option>
          <option
            value="OR"
          >
            Oregon
          </option>
          <option
            value="PA"
          >
            Pennsylvania
          </option>
          <option
            value="RI"
          >
            Rhode Island
          </option>
          <option
            value="SC"
          >
            South Carolina
          </option>
          <option
            value="SD"
          >
            South Dakota
          </option>
          <option
            value="TN"
          >
            Tennessee
          </option>
          <option
            value="TX"
          >
            Texas
          </option>
          <option
            value="UT"
          >
            Utah
          </option>
          <option
            value="VT"
          >
            Vermont
          </option>
          <option
            value="VA"
          >
            Virginia
          </option>
          <option
            value="WA"
          >
            Washington
          </option>
          <option
            value="WV"
          >
            West Virginia
          </option>
          <option
            value="WI"
          >
            Wisconsin
          </option>
          <option
            value="WY"
          >
            Wyoming
          </option>
        </select>
      </div>
    </StateSelector>
    <div
      className="LocationForm__alert-div"
      role="alert"
    >
      <ValidateState
        setStateError={[Function]}
        state=""
        stateError="You must select a state."
      >
        <p
          className="ValidateState__validator error"
          id="state-validator"
        >
          You must select a state.
        </p>
      </ValidateState>
    </div>
    <div>
      <label
        htmlFor="zipcode"
      >
        ZIP code:
         
      </label>
      <input
        aria-describedby="zipcode-validator"
        id="zipcode"
        maxLength="5"
        name="zipcode"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="LocationForm__alert-div small"
      role="alert"
    >
      <ValidateZipCode
        setZipcodeError={[Function]}
        zipcode=""
        zipcodeError="ZIP code is required and must be 5 digits in length."
      >
        <p
          className="ValidateZipCode__validator error"
          id="zipcode-validator"
        >
          ZIP code is required and must be 5 digits in length.
        </p>
      </ValidateZipCode>
    </div>
  </fieldset>
</LocationForm>
`;
